src/Layers/FeatureLayer/FeatureLayer.js

```diff
@@ -90,8 +99,23 @@ EsriLeaflet.Layers.FeatureLayer = EsriLeaflet.Layers.FeatureManager.extend({
         return;
       }
 
+      // polylines and polygons
       if (layer && layer.setLatLngs) {
-        this._updateLayerGeometry(layer, geojson);
+        this._updateLayer(layer, geojson);
+        // update custom symbology for layer if necessary
+        this.resetStyle(geojson.id);
+        return;
+      }
+
+      // points
+      if (layer && layer.setLatLng) {
```

Comment:
This doesn't check for `layer.setIcon` so if a layer has `setLatLng` but not `layer.setIcon` (like `L.CircleMarker`) it will fail.

Probably best to do a separate check entirely like `if (layer && layer.setIcon)`.